src/lexer/lex.c

```diff
@@ -528,9 +531,48 @@ static const char *ttype_names[] = {
 
 const char *ttype_name(TokenType tt) { return ttype_names[tt]; }
 
+int test_ttype_many_chars() {
+    testing_func_setup();
+
+    tassert(ttype_many_chars("foo") == TT_IDENTIFIER);
+    tassert(ttype_many_chars("struct") == TT_STRUCT);
+    tassert(ttype_many_chars("while") == TT_WHILE);
+
+    return 0;
+}
+
+int test_ttype_one_char() {
```

Comment:
Would it be possible to get some representation for failure paths, e.g. invalid tokens?

Reply:
What do you mean?

Reply:
I mean as in testing that it fails gracefully and doesn't segfault or something if we ask `ttype_string("/* This is not a token. */")` or similar.

Reply:
Yea, testing doesn't segfault, it just prints the assert that failed

Reply:
I don't mean that the testing code itself might be broken, I mean testing whether `ttype_string` might fail gracefully or just crash or something. I think this is good to test because our system should be robust.

Reply:
Ohh I see, yup we definitely should have that added. Same PR or later?

Reply:
Later is probably fine.